Document DevOps Make Commands on Cookieplone Reference Guide (Fixes #1988) (#1989)
plone/documentation: docs/reference-guide/cookieplone-make-commands.md

Reviewer: Please delete lines 14-23. They're out of sync with the content and don't actually provide anything useful.

`build-images`
:   Invokes the target `build-image` in both `backend/Makefile` and `frontend/Makefile`. 
    This builds Docker images for both backend and frontend.

## Devops

> Cookieplone projects can optionally include a `devops` folder when deployment-related options are selected during project generation, such as:
> 
> - **Add Ansible playbooks?**
> - **Add GitHub Action to Deploy this project?**

Reviewer: ```suggestion
Cookieplone projects include a `devops` folder when deployment-related questions are answered "Yes" during project generation.
These questions include the following.

-   Add Ansible playbooks?
-   Add GitHub Action to Deploy this project?
```

> When these options are enabled, Cookieplone generates a `devops/Makefile` containing deployment and CI/CD helper commands.

Reviewer: ```suggestion
When any of these options are selected, Cookieplone generates a {file}`devops/Makefile` file containing deployment and CI/CD helper commands.
```

> You can view the available DevOps commands in your generated project by running:
> 
> ```bash
> # from the project root
> make -C devops help
> 
> # or
> cd devops
> make help
> 
> ```

Reviewer: Move comments into easier to read narrative text. The syntax highlighter makes comments low-contrast.

Separate commands are easier to copy-paste with the copy-paste button.

Finally the syntax should be a generic `shell`, not a specific flavor.

````suggestion
You can view the available devops commands in your generated project by running the following help command from the project root.

```shell
make -C devops help
```

Alternatively, change the working directory to {file}`devops` and run its help command.

```shell
cd devops
make help
```
````

> The commands vary depending on your Cookieplone template options, so this is the authoritative way to inspect your environment.

Reviewer: ```suggestion
The available commands align with the selected Cookieplone template options.
```

> Below is the current working output of the DevOps make help command
> (as fixed in cookieplone-templates#305):

Reviewer: ```suggestion
When all devops options have been selected, the following output is the result of running the devops help command.
```

> ```

Reviewer: Add a syntax. Default is Python.

````suggestion
```console
````

# === DevOps Makefile Commands ===
deploy                 Deploy the project using the configured provider
deploy-check           Validate deployment configuration
deploy-ansible         Run Ansible playbooks for provisioning
deploy-rollback        Roll back the last deployment
deploy-status          Show deployment status or active revision
secrets-edit           Edit encrypted secrets
secrets-view           View decrypted secrets
lint                   Lint Ansible and DevOps configuration files
format                 Format DevOps files (YAML, JSON, etc.)
clean                  Clean temporary deployment artifacts
help                   Show this help
```

> These commands support tasks such as:
> 
> - provisioning via **Ansible**
> - deploying through **GitHub Actions** or **GitLab CI**
> - editing and viewing **encrypted secrets**
> - linting DevOps configuration
> - rolling back deployments

Reviewer: Don't overemphasize things. It's distracting.

```suggestion

These commands support devops tasks, such as:

-   provisioning via Ansible
-   deploying through GitHub Actions or GitLab CI
-   editing and viewing encrypted secrets
-   linting devops configuration
-   rolling back deployments
```

> ## Additional Documentation

Reviewer: Sentence case headings.

```suggestion
## Additional documentation
```

> The `devops` folder also includes README files generated by Cookieplone, which contain deployment instructions tailored to your setup:

Reviewer: ```suggestion
The {file}`devops` folder also includes {file}`README*.md` files generated by Cookieplone, which contain deployment instructions tailored to your setup:
```

> - {file}`devops/README.md`
> - {file}`devops/README-GHA.md` (GitHub Actions deployment)
> - {file}`devops/README-GITLAB.md` (GitLab deployment)

Reviewer: ```suggestion
-   {file}`devops/README.md`
-   {file}`devops/README-GHA.md` for GitHub Actions deployment
-   {file}`devops/README-GITLAB.md` for GitLab deployment
```

> These documents provide hands-on details such as provisioning steps, CI configuration, and environment requirements.  
> This reference section avoids duplicating that information and instead links to those files.

Reviewer: Let's combine this into one sentence. Also there are no links.

```suggestion
These documents provide details, such as provisioning steps, CI configuration, and environment requirements, and are the authoritative source of documentation for devops.
```
